modular_zubbers/code/modules/rr_opt_in/code/mind.dm

```diff
@@ -0,0 +1,84 @@
+/// If a player has any of these enabled, they are forced to use a minimum of RR_OPT_IN_ENABLED_LEVEL round removal optin. Dynamic - checked on the fly, not cached.
+GLOBAL_LIST_INIT(rr_optin_forcing_midround_antag_categories, list(
+	ROLE_CHANGELING_MIDROUND,
+	ROLE_MALF_MIDROUND,
+	ROLE_OBSESSED,
+	ROLE_SLEEPER_AGENT,
+))
+
+/// If a player has any of these enabled ON SPAWN, they are forced to use a minimum of RR_OPT_IN_ENABLED_LEVEL round removal optin for the rest of the round.
+GLOBAL_LIST_INIT(rr_optin_forcing_on_spawn_antag_categories, list(
+	ROLE_BROTHER,
+	ROLE_CHANGELING,
+	ROLE_CULTIST,
+	ROLE_HERETIC,
+	ROLE_MALF,
+	ROLE_OPERATIVE,
+	ROLE_TRAITOR,
+	ROLE_WIZARD,
+	ROLE_ASSAULT_OPERATIVE,
+	ROLE_CLOWN_OPERATIVE,
+	ROLE_NUCLEAR_OPERATIVE,
+	ROLE_HERETIC_SMUGGLER,
+	ROLE_PROVOCATEUR,
+	ROLE_STOWAWAY_CHANGELING,
+	ROLE_SYNDICATE_INFILTRATOR,
+))
+
+/datum/mind
+	/// The optin level set by preferences.
+	var/ideal_opt_in_level = RR_OPT_LEVEL_DEFAULT
+	/// Set on mind transfer. Set by on-spawn antags (e.g. if you have traitor on and spawn, this will be set to RR_OPT_LEVEL_ANTAG and cannot change)
+	var/on_spawn_rr_opt_in_level = RR_OPT_OUT
```

Comment:
Ditto
```suggestion
	/// Set on mind transfer. Set by on-spawn antags (e.g. if you have traitor on and spawn, this will be set to RR_OPT_LEVEL_ANTAG and cannot change)
	var/round_removal_allowed = FALSE
```